src/Config.cpp

```diff
@@ -437,16 +437,52 @@ bool getStatusConfig()
 // ************************************************************
 void generateSerial(bool force)
 {
-    MFeeprom.read_block(MEM_OFFSET_SERIAL, serial, MEM_LEN_SERIAL);
-    if (!force && serial[0] == 'S' && serial[1] == 'N')
+    // A serial number according old style is already generated and saved to the eeprom
+    // So keep it to avoid a connector message with orphaned board
+    if (!force && MFeeprom.read_byte(MEM_OFFSET_SERIAL) == 'S' && MFeeprom.read_byte(MEM_OFFSET_SERIAL + 1) == 'N') {
```

Comment:
All of the if statements in this method start with `!force`, which should get refactored out into a standalone `if (force)` test at the start of the function to do whatever force does (which as far as I can tell is nothing?). This will make for easier to follow function logic and simpler conditions on the rest of the if statements.